### ISSUE TITLE
The overlay doesn't go away when 'back' button is pressed. 

### ISSUE THREAD

**Issue by wongni** [RafaelBarbosatec/tutorial_coach_mark#13]
While showing tutorial overlay, if the back button is pressed, the overlay is supposed to go away but it's not.

**Comment by RafaelBarbosatec** [comment]
Good remark! I will be evaluating how to do this. if you have a suggestion I would appreciate if you opened a PR.

**Comment by CiriousJoker** [comment]
Any updates on this?

**Comment by yasinarik** [comment]
Yes, on Android, instead of acting like there is no coachmark, the back button should either show the previous coachmark step OR close the coachmark completely.

Now, let's assume that we have only one coachmark step. And when pressed the built-in back button of Android devices, it navigates back instead of closing it.

I think this is still a valid issue. Any news? Thanks a lot!!!

**Comment by yasinarik** [comment]
Thanks a lot to @masewo for the solution!!!

The below forks of this package have the `back button interceptor`. And works as expected.

If you are using Dart 2.17.0 on your project: (it uses: `back_button_interceptor: ^6.0.1`)
````yaml
  tutorial_coach_mark:
    git: https://github.com/masewo/tutorial_coach_mark 
````


Also, if you are using Dart 2.12.0 on your project: (it uses `back_button_interceptor: ^5.0.1`)
````yaml
  tutorial_coach_mark:
    git: https://github.com/yasinarik/tutorial_coach_mark 
````

**Comment by bazl-E** [comment]
> Thanks a lot to @masewo for the solution!!!
> 
> The below forks of this package have the `back button interceptor`. And works as expected.
> 
> If you are using Dart 2.17.0 on your project: (it uses: `back_button_interceptor: ^6.0.1`)
> 
> ```yaml
>   tutorial_coach_mark:
>     git: https://github.com/masewo/tutorial_coach_mark 
> ```
> 
> Also, if you are using Dart 2.12.0 on your project: (it uses `back_button_interceptor: ^5.0.1`)
> 
> ```yaml
>   tutorial_coach_mark:
>     git: https://github.com/yasinarik/tutorial_coach_mark 
> ```

Thank you this one works.its better to make a pr to this project

**Comment by vrman** [comment]
Any update  on this ?

**Comment by RafaelBarbosatec** [comment]
The overlay is a plain `OverlayEntry` inserted in the root overlay — it doesn't react to the system back button by itself. To close the tutorial on back, wrap your screen in a `PopScope` and call `tutorialMark.finish()` (or `skip()`) when the pop is intercepted. Also note that `disableBackButton: true` exists if you want the opposite (block the back button while the tutorial is active). Closing as answered.
